opta/utils/visit.py

```diff
@@ -0,0 +1,167 @@
+from collections.abc import Iterable, MutableMapping, MutableSequence
```

Comment:
another critical class that you can add a few unit tests - to help with having the team understanding and enable us to make some changes

Reply:
Adding in a separate PR to avoid blocking this PR.